### PR TITLE
Address Reza's comment

### PR DESCRIPTION
**Pull request by galiacheng**
This pull request includes various changes to the `spring-boot` and `tomcat` modules, focusing on updating dependencies, Docker configuration, and code refactoring to use Jakarta EE. The most important changes include modifying the `pom.xml` files, updating Docker configurations, and refactoring code to replace Spring annotations with Jakarta EE annotations.

### Dependency Updates:
* [`spring-boot/pom.xml`](diffhunk://#diff-107ca807a36bf4b8d6915cd626e51027067468ab5bde96316682ae9bbc6f973bL17-R17): Changed packaging from `war` to `jar`, added Jakarta EE CDI dependency, set the final name to `todo`, and configured the Spring Boot Maven plugin to repackage the application. [[1]](diffhunk://#diff-107ca807a36bf4b8d6915cd626e51027067468ab5bde96316682ae9bbc6f973bL17-R17) [[2]](diffhunk://#diff-107ca807a36bf4b8d6915cd626e51027067468ab5bde96316682ae9bbc6f973bR28) [[3]](diffhunk://#diff-107ca807a36bf4b8d6915cd626e51027067468ab5bde96316682ae9bbc6f973bR38-R44) [[4]](diffhunk://#diff-107ca807a36bf4b8d6915cd626e51027067468ab5bde96316682ae9bbc6f973bR66) [[5]](diffhunk://#diff-107ca807a36bf4b8d6915cd626e51027067468ab5bde96316682ae9bbc6f973bL76-R85) [[6]](diffhunk://#diff-107ca807a36bf4b8d6915cd626e51027067468ab5bde96316682ae9bbc6f973bR140-R146)
* [`tomcat/pom.xml`](diffhunk://#diff-e2712ba9c2992cca6dbf4b9808bcd56985b913359925d036707ade690b1c1212R28): Added Jakarta EE CDI dependency. [[1]](diffhunk://#diff-e2712ba9c2992cca6dbf4b9808bcd56985b913359925d036707ade690b1c1212R28) [[2]](diffhunk://#diff-e2712ba9c2992cca6dbf4b9808bcd56985b913359925d036707ade690b1c1212R38-R44)

### Docker Configuration:
* [`spring-boot/Dockerfile`](diffhunk://#diff-2ad4832bea4f59fa509081d1241da403734365c042d38450a710da18be37fd5cR1-R4): Added Dockerfile to build the Spring Boot application using OpenJDK 17 and exposed port 8080.
* [`spring-boot/docker-compose.yaml`](diffhunk://#diff-424b3f2c7adf2da4c51e2cc0e901272e763bc04c90c8d5b913e4e49b7486b212R13-R25): Added configuration for the Spring Boot application, including environment variables and dependencies on the PostgreSQL service.

### Code Refactoring:
* [`spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java`](diffhunk://#diff-f7d77e02ca0a7dee730b6c9fd58b4f35d1d41d53abbb4e253db77784e98f4c5bL3-R16): Replaced Spring annotations with Jakarta EE annotations (`@Autowired` to `@Inject` and added `@Transactional`).
* [`spring-boot/src/main/java/com/azure/samples/todo/ToDoItemRepository.java`](diffhunk://#diff-bc18640337e90fe5139e25a4c741f6f014165729597a849e97e933136ef1b2edL6-R6): Minor formatting changes and ensured the `@Repository` annotation is present.
* [`spring-boot/src/main/java/com/azure/samples/todo/ToDoResource.java`](diffhunk://#diff-6a7a8a6ea0874e0d62f9806e8716ecf4629a49dc7390588e94c30631c1cddd8eL3-R4): Replaced Spring annotations with Jakarta EE annotations (`@Autowired` to `@Inject`). [[1]](diffhunk://#diff-6a7a8a6ea0874e0d62f9806e8716ecf4629a49dc7390588e94c30631c1cddd8eL3-R4) [[2]](diffhunk://#diff-6a7a8a6ea0874e0d62f9806e8716ecf4629a49dc7390588e94c30631c1cddd8eL14-R14) [[3]](diffhunk://#diff-6a7a8a6ea0874e0d62f9806e8716ecf4629a49dc7390588e94c30631c1cddd8eL26)
* [`tomcat/src/main/java/com/azure/samples/todo/DefaultToDoService.java`](diffhunk://#diff-fad81123f7633fd2d138c6defd5e49f0a1efaa7a3d816fd0475d15ccc3d7ec1eL3-R16): Replaced Spring annotations with Jakarta EE annotations (`@Autowired` to `@Inject` and added `@Transactional`).
* [`tomcat/src/main/java/com/azure/samples/todo/ToDoItemRepository.java`](diffhunk://#diff-bc18640337e90fe5139e25a4c741f6f014165729597a849e97e933136ef1b2edL6-R6): Minor formatting changes and ensured the `@Repository` annotation is present.
* [`tomcat/src/main/java/com/azure/samples/todo/ToDoResource.java`](diffhunk://#diff-38b41ea9dc97af5b7767a5ee7db74d789de09da688099cb34e9335f54f4f7b3fL3-R4): Replaced Spring annotations with Jakarta EE annotations (`@Autowired` to `@Inject`). [[1]](diffhunk://#diff-38b41ea9dc97af5b7767a5ee7db74d789de09da688099cb34e9335f54f4f7b3fL3-R4) [[2]](diffhunk://#diff-38b41ea9dc97af5b7767a5ee7db74d789de09da688099cb34e9335f54f4f7b3fL14-R14)

### Documentation Update:
* [`spring-boot/README.md`](diffhunk://#diff-aaec76988f06f6311851302319118d822021737060bbf3c92269ca748b82c572L10-R20): Updated instructions to build the application using Maven and clarified Docker Compose usage.

### CODE REVIEW

File: spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 import org.springframework.stereotype.Service;
+import jakarta.transaction.Transactional; 
+import jakarta.inject.Inject;
 
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 @Service
+@Transactional
```
Comment (m-reza-rahman) [inline spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java]: I don’t think Spring recognizes this. You’ll need to use the Spring annotation.

File: jakartaee/src/main/java/com/azure/samples/todo/ToDoResource.java [hunk 2]
```diff
@@ -36,7 +36,6 @@ public ToDoItem create(@Valid ToDoItem item) {
     public void edit(
             @PathParam("id") Long id,
             @Valid ToDoItem item) {
-        item.setId(id);
```
Comment (m-reza-rahman) [inline jakartaee/src/main/java/com/azure/samples/todo/ToDoResource.java:39]: Please do this for all the versions of this code and also please remove the path parameter.

File: spring-boot/src/main/resources/application.properties [hunk 3]
```diff
@@ -6,7 +6,6 @@ spring.datasource.password=${POSTGRESQL_DB_PASSWORD:admin}
 spring.datasource.driver-class-name=org.postgresql.Driver
 
 spring.jpa.hibernate.ddl-auto=create-drop
```
Comment (m-reza-rahman) [inline spring-boot/src/main/resources/application.properties]: Please use the Jakarta Persistent property instead.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please address review comments?

File: spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java (hunk 1 above)
Comment (galiacheng) [reply spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java]: Yes, we have to add @EnableTransactionManagement to the application to enable it. 
jakarta.transaction.Transactional can be recognized by Spring, see https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/annotations.html

Comment (galiacheng) [reply spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java]: Spring Boot will detect spring-jdbc on the classpath and postgresql and will create a DataSource and a JdbcTemplate automatically. See https://spring.io/guides/gs/managing-transactions
Spring can recognize jakarta.transaction.Transactional, see https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/annotations.html

Comment (galiacheng) [reply spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java]: Here list some debug log:
```
2024-12-31T15:30:48.752+08:00  INFO 25449 --- [Todo] [nio-8080-exec-8] c.azure.samples.todo.DefaultToDoService  : Getting all items
2024-12-31T15:30:48.788+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-8] o.s.t.i.TransactionInterceptor           : Getting transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll]
Hibernate: select tdi1_0.id,tdi1_0.completed,tdi1_0.description from to_do_item tdi1_0
2024-12-31T15:30:48.960+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-8] o.s.t.i.TransactionInterceptor           : Completing transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll]
2024-12-31T15:30:55.445+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Getting transaction for [com.azure.samples.todo.DefaultToDoService.addToDoItem]
2024-12-31T15:30:55.448+08:00  INFO 25449 --- [Todo] [nio-8080-exec-2] c.azure.samples.todo.DefaultToDoService  : Adding item: ToDoItem[id=null, desciption=REST APITRE, completed=false ]
2024-12-31T15:30:55.449+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Getting transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
Hibernate: select nextval('to_do_item_seq')
2024-12-31T15:30:55.475+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Completing transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
2024-12-31T15:30:55.475+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Completing transaction for [com.azure.samples.todo.DefaultToDoService.addToDoItem]
Hibernate: insert into to_do_item (completed,description,id) values (?,?,?)
2024-12-31T15:30:59.423+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Getting transaction for [com.azure.samples.todo.DefaultToDoService.updateToDoItem]
2024-12-31T15:30:59.423+08:00  INFO 25449 --- [Todo] [nio-8080-exec-4] c.azure.samples.todo.DefaultToDoService  : Updating item: ToDoItem[id=1, desciption=, completed=false ]
2024-12-31T15:30:59.424+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Getting transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
Hibernate: select tdi1_0.id,tdi1_0.completed,tdi1_0.description from to_do_item tdi1_0 where tdi1_0.id=?
2024-12-31T15:30:59.445+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Completing transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
2024-12-31T15:30:59.445+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Completing transaction for [com.azure.samples.todo.DefaultToDoService.updateToDoItem]
```

File: jakartaee/src/main/java/com/azure/samples/todo/ToDoResource.java (hunk 2 above)
Comment (galiacheng) [reply jakartaee/src/main/java/com/azure/samples/todo/ToDoResource.java:39]: Done.

File: spring-boot/src/main/resources/application.properties (hunk 3 above)
Comment (galiacheng) [reply spring-boot/src/main/resources/application.properties]: Done. See:

- https://www.baeldung.com/spring-data-jpa-generate-db-schema
- https://jakarta.ee/specifications/persistence/3.2/jakarta-persistence-spec-3.2

**Pull request merged**